### PR TITLE
add four file

### PR DESCRIPTION
**Pull request by zzsocool**
finish all lesson03 exercises and assignments

### CODE REVIEW

File: students/Z_shen/lesson03/list.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+
+'''Series 1'''
+
+def print_current_list(list):
+    return print('The Current Fruit List is',list)
+
+def user_fruit_input():
+    return input('Enter a New Fruit into the List: ').title()
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:8]: Good use of title()

File: students/Z_shen/lesson03/list.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+def user_number_input():
+    return int(input('Enter a number between 1 to {}: '.format(len(fruit))))
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:11]: Nice use of len()!

File: students/Z_shen/lesson03/list.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+def display_number_fruit(number):
+    return print(fruit[number-1])
+fruit=['Apples','Pears','Oranges','Peaches']
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:15]: Missing spaces around assignment operator here and below

File: students/Z_shen/lesson03/list.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+print_current_list(fruit)
+
+fruit.append(user_fruit_input())
+
+print_current_list(fruit)
+
+display_number_fruit((user_number_input()))
+
+print('-'*80)
+fruit= ['Grapes'] + fruit  #Add another fruit to the beginning of the list using “+” and display the list.
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:29]: Missing space after fruit

File: students/Z_shen/lesson03/list.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+print('Add Grapes to the beginning of the list using “+” and display the list')
+print_current_list(fruit)
+
+print('-'*80)
+fruit.insert(0,'Papaya') #Add another fruit to the beginning of the list using insert() and display the list.
+print('Add Papaya to the beginning of the list using insert() and display the list.')
+print_current_list(fruit)
+
+print('-'*80)
+P_fruit=[]
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:39]: Need spaces around assignment operator

Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:39]: This should be snake_case

File: students/Z_shen/lesson03/list.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    elif answer.upper() == 'NO':
+        print('Fruit remove from your list')
+
+    else:
+        print("------------------")
+        print('Please Enter Yes or No only!!!!')
+        answer = input('Do you like fruit {}? '.format(item.lower()))
+
+print_current_list(fav_fruit)
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/list.py:98]: Good!

File: students/Z_shen/lesson03/mailroom.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+def print_donor_report(donor):
+    row='| {name:20} | {sign_1:1}  {amount:>15} | {num_gifts:^10} | {sign_2:1} {avg_gift:<15,.2f} |'.format
+
+    print('| {:20} | {:1}  {:^15} | {:^10} | {:1} {:<15} |'.format('Donor Name', ' ' ,'Total Given','Num Gifts','','Average Gift'))
+    print('-'*78)
+    for p in donor:
+        print(row(name=p[0],sign_1='$',amount=p[1], num_gifts=p[2],sign_2='$',avg_gift=int(p[1])/int(p[2])))
+
+def add_new_donor(name,amount,times,donor):
+    new_donor=[name,amount,times]
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/mailroom.py:24]: Missing spaces around assignment operator - here and elsewhere in this code

File: students/Z_shen/lesson03/mailroom.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        donor_name.append(name[0])
+    return donor_name
+
+def print_donor_report(donor):
+    row='| {name:20} | {sign_1:1}  {amount:>15} | {num_gifts:^10} | {sign_2:1} {avg_gift:<15,.2f} |'.format
+
+    print('| {:20} | {:1}  {:^15} | {:^10} | {:1} {:<15} |'.format('Donor Name', ' ' ,'Total Given','Num Gifts','','Average Gift'))
+    print('-'*78)
+    for p in donor:
+        print(row(name=p[0],sign_1='$',amount=p[1], num_gifts=p[2],sign_2='$',avg_gift=int(p[1])/int(p[2])))
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/mailroom.py:21]: Donors should be listed by total given in descending order, including new donors and donations

File: students/Z_shen/lesson03/mailroom.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                times=1
+                add_new_donor(thx_name, thx_amount, times, donor)
+                break
+        else:
+            thx_name=user_input
+            count_times(thx_name, donor)
+            total_given(thx_name, thx_amount, donor)
+            break
+    print('\nEMAIL CONTENT: ')
+    print('{}, Thank you for your generous donation of ${:,.2f} to us.\n'.format(thx_name,float(thx_amount)))
```
Comment (charlesrobison) [inline students/Z_shen/lesson03/mailroom.py:55]: Missing 'list' option to see existing donors

**Review by charlesrobison — APPROVED** [review]
Overall looking good.  Mailroom is missing a couple of elements.  Also, recommend adding a linter to your text editor to catch PEP8 syntax issues.  Please let me know if you have questions.

Ready to merge

**Comment by zzsocool** [comment]
add 'list' option to see existing donors

donors are listed by total given in descending order

**Pull request merged**